src/index.js

```diff
@@ -14,17 +14,28 @@
  *    limitations under the License.
  */
 
+import cookie from "cookie";
+import { v4 } from "uuid";
 import {
   createInstance,
   enums as OptimizelyEnums,
 } from "@optimizely/optimizely-sdk/dist/optimizely.lite.es";
-import { getDatafile, dispatchEvent } from "./optimizely_helper";
+import {
+  getDatafile,
+  dispatchEvent,
+} from "./optimizely_helper";
 
 const FASTLY_CLIENT_ENGINE = "javascript-sdk/fastly";
+const OPTIMIZELY_USER_ID_COOKIE_NAME = "optimizely_user_id";
 
 addEventListener("fetch", (event) => event.respondWith(handleRequest(event)));
 
 async function handleRequest(event) {
+  const cookies = cookie.parse(event.request.headers.get("Cookie") || '');
+
+  // Fetch user Id from the cookie if available to make sure that a returning user from same browser session always sees the same variation.
+  const userId = cookies[OPTIMIZELY_USER_ID_COOKIE_NAME] || v4();
```

Comment:
I think this is fine for now, though one corner case we might need to keep a mental note on is if `v4()` fails for whatever reason in the future (i.e. breaking change to the library).